source/utilities/imgcat

```diff
@@ -86,6 +86,29 @@ while [ $# -gt 0 ]; do
     -p|--p|--print)
         print_filename=1
         ;;
+	-u|--u|--url)
+        if curl --output /dev/null --silent --head --fail "$2"; then # check if the url is downloadable
```

Comment:
It's kind of ugly to make two requests. I haven't tested this, but I think something along these lines could be made to work:

```
encoded_image=$(curl -s "$2" | base64) || (error "No such file or url $2"; exit 2)
has_stdin=f
print_image "$file" 1 "$encoded_image" "$print_filename"
etc.
```

Reply:
I found the ugly is from indentations. I will check it out.
Moreover, your code is very precise and elegant. I will change it as well.

Reply:
set -- ${@:1:1} "-u" ${@:3}
The code is to shift all the argument without $1 and $2.
"-u" replaces $1 in order to make sure while loop will keep running -u option.

if [ "$#" -eq 2 ]; then
        exit
fi
The code is to make sure no url needed to printf picture.

To sum up, this four lines of code is to printf multiple url pictures.